src/Microsoft.Health.Abstractions/Errors/IErrorMessageService.cs

```diff
@@ -0,0 +1,22 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Health.Abstractions.Errors
+{
+    public interface IErrorMessageService
+    {
+        /// <summary>
+        /// Writes a collection of ErrorMessages to a destination
+        /// </summary>
+        /// <param name="errorMessages">A collection of error message objects.</param>
+        /// <param name="cancellationToken">A cancellation token.</param>
+        /// <returns><see cref="Task"/>Returns the successfully stored error messages.</returns>
```

Comment:
Typically the comment for an asynchronous result looks like the following:

```csharp
/// <returns>A task that represents the asynchronous operation.</returns>
```

Furthermore, if the result is `Task<T>` it does on to describe the result, then a reference is made to the result

```csharp
/// <returns>
/// A task that represents the asynchronous operation.
/// The value of the <see cref="Task{TResult}.Result"/> property contains ...
/// </returns>
```
OR

```csharp
/// <returns>
/// A task that represents the asynchronous operation.
/// The value of the <c>TResult</c> parameter contains ...
/// </returns>
```

See something like [`Stream.ReadAsync`](https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.readasync?view=net-6.0#system-io-stream-readasync(system-byte()-system-int32-system-int32)) as an example.